doc/source/api/core/typing.rst

```diff
@@ -0,0 +1,29 @@
+Typing
+======
+.. currentmodule:: pyvista
+
+Type aliases and type variable used by PyVista.
+
+
+Numeric Array-Like Types
+------------------------
+
+.. autotypevar:: NumberType
+
+.. autodata:: ArrayLike
```

Comment:
Sorry, I haven't had much time lately, so I haven't properly looked into this yet.
But another difference I noticed is the way the documentation for the type alias is generated here. For `ColorLike` it is generated using `.. autosummary::` (see utilities.rst). I'm not at all a Sphinx expert, so maybe these are equivalent, but maybe you can give this a try?
Additionally, I don't see this typing module being documented on [the deployed documentation for the 'dev' branch](https://docs.pyvista.org/version/dev/api/core/index.html)?